### PR TITLE
Add missing set operations involving `IntervalBox`es, with tests

### LINKED ISSUE #261: Add missing set operations to IntervalBox
```julia
julia> using IntervalArithmetic

julia> A = IntervalBox(1..2, 3..4)
[1, 2] × [3, 4]

julia> B = IntervalBox(0..2, 3..6)
[0, 2] × [3, 6]

julia> A ⊂ B
ERROR: MethodError: no method matching ⊂(::IntervalBox{2,Float64}, ::IntervalBox{2,Float64})
Stacktrace:
 [1] top-level scope at none:0

julia> B ⊃ A
ERROR: MethodError: no method matching ⊃(::IntervalBox{2,Float64}, ::IntervalBox{2,Float64})
Stacktrace:
 [1] top-level scope at none:0
```

### PR DESCRIPTION
**Pull request by lbenet**
Close #261

### CODE REVIEW

**Comment by lbenet** [comment]
Restarted travis, since there was some problem with one test...

**Comment by lbenet** [comment]
Tests pass. I think this is ready for review.

File: src/multidim/intervalbox.jl [hunk 1]
```diff
@@ -65,8 +65,9 @@ big(X::IntervalBox) = big.(X)
 ## set operations
 
 # TODO: Update to use generator
-⊆(X::IntervalBox{N,T}, Y::IntervalBox{N,T}) where {N,T} =
-    all(X.v .⊆ Y.v)
+⊆(X::IntervalBox{N,T}, Y::IntervalBox{N,T}) where {N,T} = all(X .⊆ Y)
+⊂(X::IntervalBox{N,T}, Y::IntervalBox{N,T}) where {N,T} = all(X .⊂ Y)
+⊃(X::IntervalBox{N,T}, Y::IntervalBox{N,T}) where {N,T} = all(X .⊃ Y)
```
Comment (Kolaru) [inline src/multidim/intervalbox.jl]: Since all definitions are the same you could use something like (untested):

```jl
for op in (:⊆, :⊂, :⊃)
    @eval $(op)(X::IntervalBox{N,T}, Y::IntervalBox{N,T}) where {N,T} = all(X .$op Y)
end
```

File: test/multidim_tests/multidim.jl [hunk 2]
```diff
@@ -37,7 +37,16 @@ let X, A  # avoid problems with global variables
...
+    @test A ⊂ B.v
+    @test A ⊂ B
+    @test A.v ⊂ B
     @test A ⊆ B.v
+    @test B ⊇ A
+    @test B.v ⊇ A
+    @test B ⊇ A.v
+    @test B ⊃ A
+    @test B.v ⊃ A
+    @test B ⊃ A.v
```
Comment (Kolaru) [inline test/multidim_tests/multidim.jl]: Adding negative tests like

```jl
@test !(B ⊂ A)
```

would be good since currently the function 

```jl
⊂(A, B) = true
```
would pass the tests.

Also there may be corner cases worth testing, like when both `IntervalBox` are the same.

Comment (lbenet) [reply test/multidim_tests/multidim.jl]: Thanks for the suggestions! I have implemented new tests as suggested, including cases where both the function acts on the same `IntervalBox`.

File: src/multidim/intervalbox.jl (hunk 1 above)
Comment (lbenet) [reply src/multidim/intervalbox.jl]: Done!

**Comment by lbenet** [comment]
Thanks @Kolaru for the review; I've followed all your recommendations, though the actual way I implemented the meta programming part is slightly different than suggested.

**Comment by dpsanders** [comment]
@lbenet Is this ready?

File: src/multidim/arithmetic.jl [hunk 3]
```diff
@@ -31,11 +31,8 @@ Base.size(X::IntervalBox{2,Float64}) = (2,)
 @inline broadcasted(f, x, y, Z::IntervalBox) = wrap(f.(x, y, Z.v))
 @inline broadcasted(f, x, Y::IntervalBox, z) = wrap(f.(x, Y.v, z))
 
-for op in (:+, :-, :∩, :∪, :⊆, :isinterior, :dot, :setdiff, :×)
+for op in (:+, :-, :∩, :∪, :⊆, :⊂, :⊃, :isinterior, :dot, :setdiff, :×)
```
Comment (dpsanders) [inline src/multidim/arithmetic.jl]: What about `⊇`?

Comment (lbenet) [reply src/multidim/arithmetic.jl]: It just works (Julia magic); see the tests. ;-)

**Comment by lbenet** [comment]
Sorry for the delay to respond; yes, I think this is ready.

**Comment by lbenet** [comment]
@dpsanders @Kolaru Any comments?

**Comment by Kolaru** [comment]
Looks good to me.

**Comment by dpsanders** [comment]
LGTM, thanks a lot!

**Comment by lbenet** [comment]
Thanks a lot!
